apis/temperature/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+#![no_std]
+
+use core::cell::Cell;
+use libtock_platform::{share, DefaultConfig, ErrorCode, Subscribe, Syscalls};
+
+pub struct Temperature<S: Syscalls>(S);
+
+impl<S: Syscalls> Temperature<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// initiate a temperature measurement used both for syncronous and asyncronous readings
+    pub fn read_temperature() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, READ_TEMP, 0, 0).to_result()
+    }
+
+    /// Register an events listener
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32,)>>,
+        subscribe: share::Handle<Subscribe<'share, S, DRIVER_NUM, 0>>,
+    ) -> Result<(), ErrorCode> {
+        S::subscribe::<_, _, DefaultConfig, DRIVER_NUM, 0>(subscribe, listener)
+    }
+
+    /// Unregister the events listener
+    pub fn unregister_listener() {
+        S::unsubscribe(DRIVER_NUM, 0)
+    }
+
+    /// initiate a syncronous temperature mesurement
+    /// Returns Ok(temperature_value) if the operation was successful
+    pub fn read_temperature_sync() -> Result<u32, ErrorCode> {
```

Comment:
```suggestion
    pub fn read_temperature_sync() -> Result<isize, ErrorCode> {
```

The temperature is reported in 100s of centigrade, meaning that 20.6 deg is 2060. The temperature can be negative, but as upcall values are always `usize`, the driver makes a typecast

https://github.com/tock/tock/blob/500caa5e882b7efe40158849fb489205f07c6a2e/capsules/src/temperature.rs#L119

that the library should reverse.